modules/ServerAPI/src/client/SaltBrowser.coffee

```diff
@@ -0,0 +1,402 @@
+				@$('.bv_structureHolder').html pngImage 
+
+				@$('.bv_editSalt').show()
+				if window.conf.salt?.editingRoles?
+					editingRoles = window.conf.salt.editingRoles.split(",")
+					if !UtilityFunctions::testUserHasRole(window.AppLaunchParams.loginUser, editingRoles)
+						@$('.bv_editSalt').hide()
+
+				@$('.bv_deleteSalt').show()
+				if window.conf.salt?.deletingRoles?
+					deletingRoles= window.conf.salt.deletingRoles.split(",")
+					if !UtilityFunctions::testUserHasRole(window.AppLaunchParams.loginUser, deletingRoles)
+						@$('.bv_deleteSalt').hide()
+			error: (result) =>
+				console.log(result)
+		)
+
+	handleCreateSaltClicked: =>
+		# Call Chemical Structure Service to Create Salt 
+		console.log("Create Salt Clicked!")
+		# Use ACASFormChemicalStructureController 
+		@$('.bv_createSalt').show()
+		@chemicalStructureController = new KetcherChemicalStructureController 
+		$('.bv_chemicalStructureForm').html @chemicalStructureController.render().el
+		# Sketcher should be controlled by CReg sketcher settings!!
+		# Need to Setup Text Boxes for Name and Abbrev
+
+	handleConfirmCreateSaltClicked: =>
+		# Need to Check Name, Struct, and Abbrev Are All Filled! 
+		saltAbbrev = UtilityFunctions::getTrimmedInput @$('.bv_abbrevName')
+		
+		saltName = UtilityFunctions::getTrimmedInput @$('.bv_saltName')
+
+		saltStruct = @chemicalStructureController.getMol()
+
+		# TO DO VALIDATION WORK HERE
+		
+		saltDict = 
+		{
+			"abbrev": saltAbbrev,
+			"molStructure": saltStruct,
+			"name": saltName,
+		}
+
+
+		$.ajax(
+			url: "/api/cmpdRegAdmin/salts",
+			type: 'POST', 
+			data: JSON.stringify(saltDict)
+			contentType: 'application/json'
+			dataType: 'json'
+			success: (result) =>
+				console.log(result)
+				@$('.bv_createSalt').hide()
+				# Reload Search
+				@searchController.doSearch("*")
+			error: (result) =>
+				console.log(result)
+				@$('.bv_createSalt').hide()
+				@$('.bv_errorCreatingSaltMessage').show()
+				# Change Windows 
+					# @$('.bv_createSalt').hide()
+					# Show Error Window! 
+					# Alternative: Insert Error Elements
+				# Report Errors To User
+		)
+
+	handleCancelCreateClicked: =>
+		console.log("Confirm Cancel Salt Clicked")
+		@$(".bv_createSalt").hide()
+
+	handleDownloadSaltClicked: =>
+		# Create Route to Download SDF of All Salts 
+		console.log("Download Salt Clicked!")
+		# Use Get Route to Obtain All Salts 
+		# See SDF Example Route Brain Bolt Gave 
+		$.ajax
+				type: 'GET'
+				url: "/api/cmpdRegAdmin/salts" 
+				success: (salts) =>
+					@downloadSDF("allSalts.txt", JSON.stringify(salts))
+				error: (result) =>
+					console.log(result)
+	
+	downloadSDF: (filename, text) => 
+		element = document.createElement('a');
+		element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(text));
+		element.setAttribute('download', filename);
+		element.style.display = 'none';
+		document.body.appendChild(element);
+		element.click();
+		document.body.removeChild(element);
```

Comment:
Is there a more proper way we want to do this? Not sure on the best practice for creating/downloading files